internal/assertions/ordered_string_array_assertion.go

```diff
@@ -0,0 +1,37 @@
+package assertions
+
+import (
+	"fmt"
+)
+
+// OrderedStringArrayAssertion : Order of the actual and expected values matters.
+// We don't alter the ordering.
+type OrderedStringArrayAssertion struct {
+	ExpectedValue []string
+	Type          string
+}
+
+func NewOrderedStringArrayAssertion(expectedValue []string, t string) OrderedStringArrayAssertion {
```

Comment:
This really doesn't make sense in the context of this challenge - feels like we just copied something over from the Redis challenge, and added `Type` to it since we needed the logs to include a string.

Why not design this properly from scratch instead? 

For example:

```
type StdoutAssertion struct {
    ExpectedLines []string
}

func NewStdoutAssertion(expectedLines []string) StdoutAssertion {}
func (a StdoutAssertion) Run(result ExecutableResult, logger Logger) (error) {}
```

(and a similar one for `StderrAssertion`)

And as things evolve we can add different types, for example if we have one test that only expects a single value we can split this into `StdoutAssertion` (for a single value) and `StdoutLinesAssertion` (for multiple lines)